### ISSUE TITLE
Users can make courses available offline

### ISSUE THREAD

**Issue by pedro16v** [catalpainternational/canoe#463]
## Overview
Content is currently automatically cached as long as there is space available in the device. Depending on the specific caching policy, some larger elements might not be cached (large images, videos). It can also happen that new content is made available on the server but is not immediately downloaded by the device. In order to make sure that users are saving content that can be used offline, we need to inform users about each course’s caching status and provide an explicit mechanism on the interface to trigger the download.

## Conditions of satisfaction
* The course screen has an option to make course available offline (download)
  * The same option should enable the reverse behaviour (to release storage)
* The user is informed if the course is already completely cached
* If the course size is larger than the available space on the device, a warning is shown to prompt the user to release space by deleting other data.

## Resources
* Engineering estimate: Days to Week

## Notes
* Front-end can jump in and wire an interface but only after the team merges and tests the “ft-offline” branch. We need Lee for this.
* We can’t currently handle exceptions such as “your storage is full”
* Currently, everything except for media is already being cached and is available offline.
* Not a priority for PeP.
